### PR TITLE
[FAB-17138] Publish docker images to nexus

### PR DESCRIPTION
**Pull request by jt-nti**
Signed-off-by: James Taylor <jamest@uk.ibm.com>

### CODE REVIEW

File: ci/azure-pipelines.yml [hunk 1]
```diff
@@ -129,47 +122,64 @@ stages:
...
-        steps:
-          - script: |
-              env | sort
-              echo "Update the version number"
-              echo "Make sure release notes are present"
-              echo "Make sure change history is present"
-      - job: npm_publish
+      - job: publish_release
         steps:
+          - template: templates/build-data.yml
```
Comment (mbwhite) [inline ci/azure-pipelines.yml]: Is this line needed here?

Comment (mbwhite) [reply ci/azure-pipelines.yml]: oh it's used lower down... is the information from the previous stage not available here?

File: ci/azure-pipelines.yml [hunk 2]
```diff
@@ -129,47 +122,64 @@ stages:
       - job: snapshot_publish
         dependsOn: main
         condition: and(succeeded(),eq(variables['Build.Reason'], 'IndividualCI'))
+        variables:
+          PACKAGE_VERSION: $[ dependencies.main.outputs['BuildData.PACKAGE_VERSION'] ]
```
Comment (mbwhite) [inline ci/azure-pipelines.yml]: we're not going to get extra spaces are in the version number   '$[ '  ?

File: ci/azure-pipelines.yml [hunk 3]
```diff
@@ -52,16 +46,15 @@ variables:
...
+#
 stages:
   - stage: Build_and_test
     jobs:
       - job: main
-        steps:
-          - script: |
-               env | sort            
+        steps:                
+          - template: templates/build-data.yml
```
Comment (mbwhite) [inline ci/azure-pipelines.yml]: ahhh so that's how things can be simplified...

File: ci/azure-pipelines.yml (hunk 1 above)
Comment (jt-nti) [reply ci/azure-pipelines.yml]: You can't share variables between stages without getting in to REST calls to the azure pipeline API as far as I can tell. A template seemed way simpler!

File: ci/azure-pipelines.yml (hunk 2 above)
Comment (jt-nti) [reply ci/azure-pipelines.yml]: I copied it from the 1.4 branch so hopefully not! https://github.com/hyperledger/fabric-chaincode-java/blob/release-1.4/ci/azure-pipelines.yml#L122

File: ci/azure-pipelines.yml (hunk 3 above)
Comment (jt-nti) [reply ci/azure-pipelines.yml]: You can even share templates between repos which might be interesting for some stuff ...in the future!

File: ci/azure-pipelines.yml (hunk 1 above)
Comment (mbwhite) [reply ci/azure-pipelines.yml]: ah well in that case - this seems like the best solution.

**Pull request merged**